addons/sale/static/src/js/sale_product_field.js

```diff
@@ -292,6 +296,9 @@ export class SaleOrderLineProductField extends ProductLabelSectionAndNoteField {
             pricelist_id: saleOrder.pricelist_id[0],
             date: serializeDateTime(saleOrder.date_order),
             edit: edit,
+            options: {
+                showQuantity: show_quantity,
+            },
```

Comment:
Why would the quantity be hidden in sale? In fact I now realise that the diff should only be in `website_sale` only, as it does not makes sense to hide the quantity fields in sale